syntax = "proto3";

package scanner.proto;

message Frame {
  int64 buffer = 1;
  int32 width = 2;
  int32 height = 3;
}

message BoundingBox {
  float x1 = 1;
  float y1 = 2;
  float x2 = 3;
  float y2 = 4;
  float score = 5;
  int32 track_id = 6;
  double track_score = 7;
  int32 label = 8;
}

message Point {
  float x = 1;
  float y = 2;
  float score = 3;
}

message NetDescriptor {
  string model_path = 1;
  string model_weights_path = 2;

  repeated string input_layer_names = 3;
  repeated string output_layer_names = 4;

  int32 input_width = 5;
  int32 input_height = 6;

  repeated float mean_colors = 7;
  // or
  repeated float mean_image = 8;
  int32 mean_width = 9;
  int32 mean_height = 10;

  bool normalize = 11;
  bool preserve_aspect_ratio = 12;
  bool transpose = 13;
  int32 pad_mod = 14;
  bool uses_python = 15;
}
